src/lib/components/embla/Embla.svelte

```diff
@@ -0,0 +1,168 @@
+
+                        if (slideIndex === loopItem.index && target !== 0) {
+                            const sign = Math.sign(target);
+
+                            if (sign === -1) {
+                                diffToTarget = scrollSnap - (1 + scrollProgress);
+                            }
+                            if (sign === 1) {
+                                diffToTarget = scrollSnap + (1 - scrollProgress);
+                            }
+                        }
+                    });
+                }
+
+                const tweenValue = 1 - Math.abs(diffToTarget * tweenFactor);
+                const scale = numberWithinRange(tweenValue, 0.8, 1).toString();
+                const tweenNode = tweenNodes[slideIndex];
+                tweenNode.style.transform = `scale(${scale})`;
+            });
+        });
+    };
+
+    const onEmblaInit = (event: CustomEvent<EmblaCarouselType>) => {
+        emblaApi = event.detail;
+
+        setTweenNodes(emblaApi);
+        setTweenFactor(emblaApi);
+        tweenScale(emblaApi);
+
+        emblaApi
+            .on('scroll', () => {
+                selectedScrollIndex = emblaApi.selectedScrollSnap();
+            })
+            .on('init', togglePrevNextBtnsState)
+            .on('select', togglePrevNextBtnsState)
+            .on('reInit', togglePrevNextBtnsState)
+            .on('reInit', setTweenNodes)
+            .on('reInit', setTweenFactor)
+            .on('reInit', tweenScale)
+            .on('scroll', tweenScale)
+            .on('slideFocus', tweenScale);
+    };
+</script>
+
+<div class="embla web-carousel">
+    {#if hasPrev}
+        <button class="web-carousel-button web-carousel-button-start" on:click={onPrev}>
+            <span class="web-icon-arrow-left" aria-hidden="true"></span>
+        </button>
+    {/if}
+    {#if hasNext}
+        <button class="web-carousel-button web-carousel-button-end" on:click={onNext}>
+            <span class="web-icon-arrow-right" aria-hidden="true"></span>
+        </button>
+    {/if}
+
+    <div class="embla__viewport" use:embla={{ options, plugins }} on:emblaInit={onEmblaInit}>
+        <ul class="embla__container">
+            <slot />
+        </ul>
+    </div>
+</div>
+
+<div class="web-carousel-bullets">
+    <ul class="web-carousel-bullets-list">
+        {#each Array.from({ length: emblaApi?.scrollSnapList().length }) as _, i}
+            <li class="web-carousel-bullets-item">
+                <button
+                    class="web-carousel-bullets-button"
+                    class:is-selected={selectedScrollIndex === i}
+                    aria-label={`gallery item ${i + 1}`}
+                    on:click={() => onSelect(i)}
+                ></button>
+            </li>
+        {/each}
+    </ul>
+</div>
+
+<style>
+    .embla {
+        overflow: hidden;
+        position: relative;
+    }
+
+    .embla__container {
+        display: flex;
+    }
+</style>
```

Comment:
feel free to use sass here, you also dont need to care about css scopes with svelte and can just use `.container` 👍🏻

Reply:
This is actually a requirement so `embla` knows which element to attach to.

https://www.embla-carousel.com/get-started/svelte/